gammapy/utils/simulate.py

```diff
@@ -0,0 +1,79 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Simulate observations"""
+import numpy as np
+import astropy.units as u
+from ..cube import MapDataset, PSFKernel
+from ..cube import make_map_exposure_true_energy, make_map_background_irf
+from ..maps import WcsNDMap
+from ..cube.models import BackgroundModel
+
+__all__ = ["simulate_3d"]
+
+
+def simulate_3d(
+    skymodel, geom, pointing, irfs, livetime=1 * u.h, offset=0 * u.deg, edisp=False
+):
+
+    """Simulate a 3D dataset
+    
+    Simulate a source defined with a sky model for a given pointing,
+    geometry and irfs for a given exposure time.
+    This will return a dataset object which includes the counts cube,
+    the exposure cube, the psf cube, the background model and the sky model.
+
+    Parameters
+    ----------
+    skymodel : `~gammapy.cube.models.SkyModel`
+        Background model map
+    geom : `~gammapy.maps.wcs.WcsGeom`
+        Geometry object for the observation
+    pointing : `~astropy.coordinates.SkyCoord`
+        Pointing position
+    irfs : 
+        Irfs used for simulating the observation
+    livetime : `~astropy.units.quantity.Quantity`
+        Livetime exposure of the simulated observation
+    offset : `~astropy.units.quantity.Quantity`
+        Offset from the center of the pointing position.
+        This is used for the PSF and Edisp estimation
+    edisp : 
+        Whether to include energy dispersion in the dataset.
+    Returns
+    ---------
+    dataset : `~gammapy.cube.fit.MapDataset`
+        A dataset of the simulated observation.
+    
+    """
+
+    background = make_map_background_irf(
+        pointing=pointing, ontime=livetime, bkg=irfs["bkg"], geom=geom
+    )
+
+    background_model = BackgroundModel(background)
+
+    psf = irfs["psf"].to_energy_dependent_table_psf(theta=offset)
+    psf_kernel = PSFKernel.from_table_psf(psf, geom, max_radius=0.4 * u.deg)
```

Comment:
This should not be a default argument. Depending on the energy range used this might be too low.
Ideally the `PSFKernel` should be able to determine the best maximal radius given the energy interval and `PSF` considered.

Reply:
Agreed.
As the automatic size determination is not available what do you suggest ?
I can change to 0.5° (0.7° to be safe ?) which will suit most of the cases I believe.
Or do you want to make that an extra argument with a default at 0.5° ?

Reply:
I think the options are to either use the containment radius or just make `max_radius` an option to the `simulate_3d` function and give a conservative default. We're using FFT convolution for the PSF and the performance only weakly depends of the size of the PSF kernel (see https://github.com/adonath/notebooks-public/blob/master/convolution_functions_comparison.ipynb).

Reply:
Ok, I will make this an argument and fix it's default size at 0.8°.